book/platform/README.md

```diff
@@ -0,0 +1,473 @@
+# The OCaml Platform
+
+So far in the book, we've gone through a number of techniques you can use to build real OCaml programs.
+We'll now wrap up this part by examining the tools you can use for editing,
+compiling, testing, documenting and publishing your own projects.
+
+The OCaml community has adopted a platform of modern tools to specify your project
+metadata and interface it with IDEs such as Visual Studio Code, to generate API
+documentation, and also to adopt modern software engineering practises such as
+continuous integration (CI) and testing.
+
+## A Hello World OCaml Project
+
+Let's start by creating a sample OCaml project and navigating around it.  opam
+provides a project generator known as `spin` that can be used to create various
+types of projects.
+
+```sh dir=examples/correct/opam-spin-ls,skip
+$ opam spin ls
+
+  bin
+    Native project containing a binary
+
+  c-bindings
+    Bindings to a C library
+
+  cli
+    Command Line Interface releasable on Opam
+
+  js
+    Javascript application with Js_of_ocaml
+
+  lib
+    Library releasable on Opam
+
+  ppx
+    PPX library
+
+```
+
+Before we dive into any of these, we'll generate a tutorial using Spin's built-in
+hello world project. 
+
+<!-- ```sh dir=examples/correct/opam-spin-hello
+TODO need a way to stop spin from running the opam install commands
+ -->
+
+```sh skip
+$ opam spin hello hello-world -vvv
+
+🏗️  Creating a new project from hello in hello-world
+Done!
+
+🎁  Installing packages globally. This might take a couple minutes.
+```
+
+Spin will create a `hello-world` directory and populate it with a skeleton
+OCaml project.  This sample project has all the metadata required for us to learn more
+about the opam package manager and the dune build tool that we've used earlier in the book.
+First use opam to install the dependencies required for our hello world project.
+
+<!-- ```sh dir=examples/correct/hello-world -->
+```
+$ make deps
+```
+
+This will invoke the opam CLI to install the project dependencies and some useful
+tools.  opam doesn't require any special user permissions and stores all of the 
+files it installs in `~/.opam` (for global installations) and `_opam` in the
+working directory for the project-local installations.  You can use `opam env`
+to add the right directories to your local shell path:
+
+```
+$ eval $(opam env)
```

Comment:
The Makefile calls the commands with `opam exec --`, so you don't need to source `opam env` to run the make commands. Hopefully, the Makefile provides all the commands the user needs to work with the project, so I don't know if we want to instruct them to run `eval $(opam env)`. Maybe it belongs to an "Advanced" section at the end of the chapter?